src/frontend/Info.ml

```diff
@@ -1,55 +1,46 @@
 open Core_kernel
 open Ast
 open Middle
+open Yojson.Basic
 
-let rec sized_basetype_dims t =
+let rec unsized_basetype_json t =
+  let base_type_dims type_ dim : t =
+    `Assoc [("type", `String type_); ("dimensions", `Int dim)] in
   match t with
-  | SizedType.SInt -> ("int", 0)
-  | SReal -> ("real", 0)
-  | SComplex -> ("complex", 0)
-  | SVector _ | SRowVector _ -> ("real", 1)
-  | SComplexVector _ | SComplexRowVector _ -> ("complex", 1)
-  | SMatrix _ -> ("real", 2)
-  | SComplexMatrix _ -> ("complex", 2)
-  | SArray (t, _) ->
-      let bt, n = sized_basetype_dims t in
-      (bt, n + 1)
-
-let rec unsized_basetype_dims t =
-  match t with
-  | UnsizedType.UInt -> ("int", 0)
-  | UReal -> ("real", 0)
-  | UComplex -> ("complex", 0)
-  | UVector | URowVector -> ("real", 1)
-  | UComplexVector | UComplexRowVector -> ("complex", 1)
-  | UMatrix -> ("real", 2)
-  | UComplexMatrix -> ("complex", 2)
-  | UArray t ->
-      let bt, n = unsized_basetype_dims t in
-      (bt, n + 1)
+  | UnsizedType.UInt -> base_type_dims "int" 0
+  | UReal -> base_type_dims "real" 0
+  | UComplex -> base_type_dims "complex" 0
+  | UVector | URowVector -> base_type_dims "real" 1
+  | UComplexVector | UComplexRowVector -> base_type_dims "complex" 1
+  | UMatrix -> base_type_dims "real" 2
+  | UComplexMatrix -> base_type_dims "complex" 2
+  | UArray t' -> (
+    match unsized_basetype_json t' with
+    | `Assoc (ty :: ("dimensions", `Int dim) :: x) ->
+        `Assoc (ty :: ("dimensions", `Int (dim + 1)) :: x)
+    | _ ->
+        Common.FatalError.fatal_error_msg
+          [%message "Failed to produce info for type " (t : UnsizedType.t)] )
```

Comment:
Bleh, destructuring just to add to the dimention? I think an internal helper function makes this cleaner.
```ocaml
let unsized_basetype_json t =
  let rec helper type_ dim =
    match type_ with
    | UReal -> base_type_dims "real" dim
    | UVector -> base_type_dims "real" (dim+1)
    | UArray t -> helper t (dim+1)
  in
  helper t 0
```